db/memtable.cc

```diff
@@ -632,8 +634,13 @@ static bool SaveValue(void* arg, const char* entry) {
   // all entries with overly large sequence numbers.
   uint32_t key_length;
   const char* key_ptr = GetVarint32Ptr(entry, entry + 5, &key_length);
+  Slice user_key_slice = Slice(key_ptr, key_length - 8);
   if (s->mem->GetInternalKeyComparator().user_comparator()->Equal(
-          Slice(key_ptr, key_length - 8), s->key->user_key())) {
+          user_key_slice, s->key->user_key()) ||
+      s->mem->GetInternalKeyComparator()
+              .user_comparator()
+              ->CompareWithoutTimestamp(user_key_slice, s->key->user_key()) ==
```

Comment:
Same as the comment somewhere else. If by default timestamp length is 0, can we just use CompareWithoutTimestamp() rather than do it twice?

Reply:
Done.